src/AddressRegistry.Consumer.Read.StreetName/Infrastructure/Program.cs

```diff
@@ -41,99 +39,185 @@ public static async Task Main(string[] args)
-
-                            var actualContainer = container.GetRequiredService<ILifetimeScope>();
+                            consumerOptions.ConfigureSaslAuthentication(new SaslAuthentication(
+                                hostContext.Configuration["Kafka:SaslUserName"],
+                                hostContext.Configuration["Kafka:SaslPassword"]));
 
-                            var bosaItemConsumerTask =
-                                new StreetNameBosaItemConsumer(actualContainer, kafkaOptions, consumerOptions)
-                                    .Start(cancellationToken);
+                            return new Consumer(consumerOptions, c.Resolve<ILoggerFactory>());
+                        })
+                        .As<IConsumer>()
+                        .SingleInstance();
 
-                            var latestItemConsumerTask =
-                                new StreetNameLatestItemConsumer(actualContainer, kafkaOptions, consumerOptions)
-                                    .Start(cancellationToken);
+                    builder
+                        .Register(c =>
+                        {
+                            var bootstrapServers = hostContext.Configuration["Kafka:BootstrapServers"];
+                            var topic = $"{hostContext.Configuration["Topic"]}" ?? throw new ArgumentException("Configuration has no AddressTopic.");
+                            var suffix = hostContext.Configuration["ConsumerGroupSuffix"];
+                            var consumerGroupId = $"{nameof(AddressRegistry)}.StreetNameLatestItemConsumer.{topic}{suffix}";
+
+                            var consumerOptions = new ConsumerOptions(
+                                new BootstrapServers(bootstrapServers),
+                                new Topic(topic),
+                                new ConsumerGroupId(consumerGroupId),
+                                EventsJsonSerializerSettingsProvider.CreateSerializerSettings());
 
-                            Log.Information("The kafka consumer streetname was started");
+                            consumerOptions.ConfigureSaslAuthentication(new SaslAuthentication(
+                                hostContext.Configuration["Kafka:SaslUserName"],
+                                hostContext.Configuration["Kafka:SaslPassword"]));
+
+                            return new IdempotentConsumer<StreetNameConsumerContext>(
+                                consumerOptions,
+                                c.Resolve<IDbContextFactory<StreetNameConsumerContext>>(),
+                                c.Resolve<ILoggerFactory>());
+                        })
+                        .As<IIdempotentConsumer<StreetNameConsumerContext>>()
+                        .SingleInstance();
+
+                    builder
+                        .RegisterModule(new DataDogModule(hostContext.Configuration));
```

Comment:
In the old code, the StreetNameConsumerModule was also registered. Is this not necessary anymore?

Reply:
Correct it's all done here.

https://github.com/Informatievlaanderen/address-registry/pull/894/files#diff-70223d4df4b9db7c2fecfe2d332299e8ce0ddcfb6ca29782d28497bee55d0341R75-R88